run.c

```diff
@@ -17,6 +17,10 @@ Then run with:
 #include <fcntl.h>
 #include <sys/mman.h>
 
+#ifndef DTYPE
```

Comment:
It's better to typedef dtype rather than spreading DTYPE macro all over.
```suggestion
#ifdef DTYPE
typedef DTYPE dtype;
else
typedef float dtype;
#else

```